Add Diversity Documentation

zanbaldwin opened the pull request:
- Create a section, under Contributing, for the Diversity Initiative.
- Document the initiative's governance, as proposed by the Diversity Initiative.

javiereguiluz commented on contributing/diversity/governance.rst:
```diff
@@ -0,0 +1,110 @@
+For an actionable item to pass, approval from greater than 50% of the voting key
+players is required. Use or management of finances/donations require at least a
+two-thirds majority to pass.
+The quorum required for votes to be considered valid is 75% of active, appointed
```
75% of 5 players = 3.75 -> how should we round this? 3 or 4? Also, if we require 4 out of 5 to vote, we could require the 5 of them.

zanbaldwin replied:
My understanding is that greater than 50% of 75% need to vote yes for something to pass.

So, if 4 people vote (2 yes, 2 no) it would be a stalemate and would not pass - the fifth member would be required to be the deciding vote.

TimoBakx replied:
Requiring 100% is not handy when we ever decide to increase the number of key players or when 1 person is not available.
In this case, 4 players voting could let an item pass with 3 votes for and 1 against.

More than 50% of the votes need to be 'yes'. More than 75% of the key players should vote.

javiereguiluz replied:
@TimoBakx I think we are talking about different things.

The document says: *"to be valid, 75% of total people must vote"*  Total people is 5, so 75% is 3.75. Should we require 3 or 4?

What I added is that if we require 4 out of 5 people to vote, is practically the same as requiring 5. But that's another point. The question is: do we require 3 or 4 people to vote? Thanks!

TimoBakx replied:
I think the wording should be changed to "to be valid, at least 75% of total people must vote".

javiereguiluz replied:
A minor note: I think the original redaction was nice and precise ... I'd just want to clarify what happens when 75% leads to a non-integer division. Thanks!

TimoBakx replied:
You're right, @javiereguiluz. Three quarters of a person is... a bad idea. ^_^
Adding the "at least" part would clarify that in either case.

zanbaldwin replied:
Added *at least 75%* to account for non-integer division :+1:

javiereguiluz commented on contributing/diversity/governance.rst:
```diff
@@ -0,0 +1,110 @@
+Voting
+~~~~~~
+
+Key players have the right to vote on proposals for actionable items.
```
We should clarify that players can only vote "yes" or "no". They cannot abstain or vote "null".

zanbaldwin replied:
I guess they could abstain, but given the quorum limits it would most likely be equivalent to a "no" vote. perhaps this should be reworded.

lsmith77 replied:
basically if all 4 people vote and one person abstains and 2 people vote in favor and one against .. then the qorum would be passed and the result would be yes with over 50%

javiereguiluz replied:
OK, but please let's be explicit about the possible valid votes of people. Thanks.

TimoBakx replied:
I don't think so. 2 out of 4 people is 50%, which is not **over** 50%. Or do you not count people that abstain? But in that case, should you count them towards the quorum?

zanbaldwin replied:
I think that the quorum should be *voting* people, and should not count abstains (same as not submitting a vote, not replying, or being unavailable). How about adding to the sentence below, something like:

> *"The quorum of "yes" or "no" votes required for a decision to be considered valid is at least 75% of active, appointment key players.

With 5 key players, and 3 options (yes, no, abstain), there are 125 different outcomes - I'll try to come up with an easy-to-digest way of presenting the outcomes of this (Reject or Pass).

lsmith77 replied:
in my scenario I would count it as 80% quorum met and 66% votes in favor and 33% opposed. if it would be 50% then abstain would be the same as no.

but sure we can make this more explicit.

lsmith77 replied:
I guess with 5 people, we could even list all possible permutations :)

zanbaldwin replied:
| Yes | No | Abstain | Result |
|-----|----|---------|--------|
| 5   | 0  | 0       | Pass   |
| 4   | 1  | 0       | Pass   |
| 4   | 0  | 1       | Pass   |
| 3   | 2  | 0       | Pass   |
| 3   | 1  | 1       | Pass   |
| 3   | 0  | 2       | Reject |
| 2   | 3  | 0       | Reject |
| 2   | 2  | 1       | Stalemate (Reject) |
| 2   | 1  | 2       | Reject |
| 2   | 0  | 3       | Reject |
| 1   | 4  | 0       | Reject |
| 1   | 3  | 1       | Reject |
| 1   | 2  | 2       | Reject |
| 1   | 1  | 3       | Reject |
| 1   | 0  | 4       | Reject |
| 0   | 5  | 0       | Reject |
| 0   | 4  | 1       | Reject |
| 0   | 3  | 2       | Reject |
| 0   | 2  | 3       | Reject |
| 0   | 1  | 4       | Reject |
| 0   | 0  | 5       | Reject |

TimoBakx replied:
People that don't vote get an automatic "abstain"?

zanbaldwin replied:
Those results permutations listed above are for an abstain vote **not** counting towards the quorum.

It would add more complexity to count an abstained vote towards the quorum, as there'd be 4 options (yes, no, abstain, no reply and not counted towards quorum), though I'm not sure on the best course of action.

TimoBakx replied:
To me, this feels a bit weird in this scenario:
3 votes for, 1 abstain and 1 yet to vote... If that last person picks "abstain", the vote will be rejected. However, if that person picks "no", the vote will pass.

TimoBakx replied:
Unless players are only allowed to vote "yes" or "no". Then it would make sense, since you can't vote with 3 players or less.

zanbaldwin replied:
Yeah that is a bit weird. I think that an abstain vote is the same as not voting, ie. people can *only* vote **yes** or **no**.

zanbaldwin replied:
Okay, following this discussion I've updated the document for clarification.

javiereguiluz commented on contributing/diversity/governance.rst:
```diff
@@ -0,0 +1,110 @@
+The project leader, Fabien Potencier, is responsible for publicly appointing
+five (5) key players of the initiative to provide guidance and drive it forward,
+but also retains the right to revoke any of the key players at any time. These
+guiding key players should:
```
Should we require a certain amount of diversity for the members of this group?

zanbaldwin replied:
That would be nice to have and I think the initiative leans in that direction already - but requiring that would mean having to quantify it somehow... How would we measure that diversity, and what or how many minorities should we enforce having?

fabpot replied:
I would indeed not introduce "quotas".

javiereguiluz commented on contributing/diversity/governance.rst:
```diff
@@ -0,0 +1,110 @@
+rest of the community. Key players should lead by example to create a culture
+that enables members to feel it is safe to suggest, question and challenge -
+rather than avoid - difficult ideas and topics. Key players guide the initiative
+in being transparent, accountable and open.
```
About transparency and accountability, could we mention explicitly above that all discussions must be done in publicly accessible mediums, such as public channels on Symfony's Slack?

zanbaldwin replied:
That's probably a good idea, but I'm not sure about enforcing it for certain things (an example would be Sage Sharp consulting which was discussed in today's meeting).
/cc @lsmith77

lsmith77 replied:
also its simply not practical .. there can be DMs, there could even be calls or discussions at conferences. it could say that all discussions should be public as much as possible .. and of course we could mandate a public debate before a vote.

javiereguiluz commented on contributing/diversity/index.rst:
```diff
@@ -0,0 +1,7 @@
+Community
```
Maybe change the title of this page to *"Symfony Diversity"* or *"Symfony Diversity & Inclusion"* or *"Symfony Diversity & Inclusion Initiative"*?

zanbaldwin replied:
Sorry, was using `/contributing/community/index.rst` as a template for the formatting and unconsciously retyped *Community*!

zanbaldwin replied:
I've named it *Diversity Initiative* (rather than *Symfony Diversity Initiative*) to follow the convention of similar titles *Code of Conduct* and *Community*.

javiereguiluz reviewed (commented):
Thanks for contributing this doc. I like it a lot and, since it's written by a native English speaker, it's fantastically redacted. Thanks!

zanbaldwin commented on contributing/map.rst.inc:
```diff
@@ -33,3 +33,7 @@
   * :doc:`Community Reviews </contributing/community/reviews>`
   * :doc:`Other Resources </contributing/community/other>`
+
+* **Diversity**
```
Should this be just *Diversity* or *Diversity Initiative*?

fabpot replied:
Diversity sounds good to me.

zanbaldwin committed:
Add Diversity Documentation

Create a section, under Contributing, for the Diversity Initiative.
Document the initiative's governance.

zanbaldwin commented:
:+1: all good to merge, thanks!

javiereguiluz commented:
Merged! Zan, thanks for contributing this ... and thanks to reviewers too!
